sql/rpl_constants.h

```diff
@@ -63,4 +67,126 @@ enum ExtraRowInfoFormat {
   ERIF_MULTI        =  255
 };
 
+static const size_t ENCRYPTION_MASTER_KEY_NAME_MAX_LEN = 100;
+static const size_t ENCRYPTION_SERVER_UUID_LEN = 36;
+static const size_t ENCRYPTION_KEY_LEN = 32;
+
+#define BINLOG_CRYPTO_SCHEME_LENGTH 1
+#define BINLOG_KEY_VERSION_LENGTH   4
+#define BINLOG_IV_LENGTH            MY_AES_BLOCK_SIZE
+#define BINLOG_IV_OFFS_LENGTH       4
+#define BINLOG_NONCE_LENGTH         (BINLOG_IV_LENGTH - BINLOG_IV_OFFS_LENGTH)
+
+struct Binlog_crypt_data {
+  uint  scheme;
+  uint  key_version, key_length, ctx_size;
+  uchar *key;
+  uchar nonce[BINLOG_NONCE_LENGTH];
+  uint dst_len;
+  uchar iv[BINLOG_IV_LENGTH];
+
+  Binlog_crypt_data()
+    : scheme(0)
+    , key(NULL) 
+  {}
+
+  void free_key()
+  {
+    if (key != NULL)
+    {
+      DBUG_ASSERT(key_length == 16);
+      memset_s(key, 512, 0, key_length);
+      my_free(key);
+      key= NULL;
+      key_length= 0;
+    }
+  }
+
+  ~Binlog_crypt_data()
+  {
+    free_key();
+  }
+
+  Binlog_crypt_data& operator=(const Binlog_crypt_data &b)
```

Comment:
Interesting assignment semantics in case b.scheme != 1, and what about the copy constructor?

Reply:
Changed.